tests/integration/test_actions.py

```diff
@@ -0,0 +1,133 @@
+
+type TwoHosts = tuple[Host, Host]
+
+
+pytestmark = [pytest.mark.asyncio]
+
+async def is_success(job: Job) -> bool:
+    return await job.get_status() == "success"
+
+async def is_aborted(job: Job) -> bool:
+    return await job.get_status() == "aborted"
+
+async def is_running(job: Job) -> bool:
+    return await job.get_status() == "running"
+
+def build_name_mapping(*iterables: Iterable[MappingPair]) -> set[tuple[str, str, str]]:
+    return {(c.service.name, c.name, h.name) for c, h in chain.from_iterable(iterables)}
+
+@pytest_asyncio.fixture()
+async def cluster_bundle(adcm_client: ADCMClient, tmp_path: Path) -> Bundle:
+    bundle_path = pack_bundle(from_dir=BUNDLES / "complex_cluster", to=tmp_path)
+    return await adcm_client.bundles.create(source=bundle_path, accept_license=True)
+
+
+@pytest_asyncio.fixture()
+async def hostprovider_bundle(adcm_client: ADCMClient, tmp_path: Path) -> Bundle:
+    bundle_path = pack_bundle(from_dir=BUNDLES / "simple_hostprovider", to=tmp_path)
+    return await adcm_client.bundles.create(source=bundle_path, accept_license=True)
+
+
+@pytest_asyncio.fixture()
+async def cluster(adcm_client: ADCMClient, cluster_bundle: Bundle) -> Cluster:
+    cluster = await adcm_client.clusters.create(bundle=cluster_bundle, name="Awesome Cluster")
+    await cluster.services.add(filter_=Filter(attr="name", op="eq", value="with_host_actions"))
+    return cluster
+
+
+@pytest_asyncio.fixture()
+async def hosts(adcm_client: ADCMClient, hostprovider_bundle: Bundle) -> TwoHosts:
+    hp = await adcm_client.hostproviders.create(bundle=hostprovider_bundle, name="Awesome HostProvider")
+    coros = (adcm_client.hosts.create(hostprovider=hp, name=f"host-{i+1}") for i in range(2))
+    await asyncio.gather(*coros)
+    hosts = await adcm_client.hosts.all()
+    return tuple(hosts)  # type: ignore[reportReturnType]
+
+
+async def test_run_action_with_mapping_and_config(adcm_client: ADCMClient, cluster: Cluster, hosts: TwoHosts) -> None:
+    mapping = await cluster.mapping
+
+    assert len(mapping.all()) == 0
+    assert len(await mapping.hosts.all()) == 0
+    assert len(await mapping.components.all()) == 2
+
+    await cluster.hosts.add(host=hosts)
+    host_1, host_2 = await mapping.hosts.all()
+
+    service_1 = await cluster.services.get(display_name__eq="with_host_actions")
+    component_1_s1 = await service_1.components.get(name__eq="c1")
+    component_2_s1 = await service_1.components.get(name__eq="c2")
+
+    await mapping.add(component=component_1_s1, host=(host_1, host_2))
+    assert len(tuple(mapping.iter())) == 2
+    await mapping.save()
+
+    ## run action host_action_config_hc_acl
+
+    host_action = await host_1.actions.get(name__eq="host_action_config_hc_acl")
+
+    task_mapping = await host_action.mapping
+    await task_mapping.remove(component=component_1_s1, host=host_1)
+    await task_mapping.add(component=component_2_s1, host=host_1)
+
+    task_config = await host_action.config
```

Comment:
There's no tasks in here, all `action`s

Client doesn't use term `Task` at all, so please use "action" based naming